server/src/domain/library/library.service.ts

```diff
@@ -129,27 +141,28 @@ export class LibraryService extends EventEmitter {
           if (matcher(path)) {
             await this.scanAssets(library.id, [path], library.ownerId, false);
           }
-          this.emit('add', path);
+          this.emit(StorageEventType.ADD, path);
         },
         onChange: async (path) => {
           this.logger.debug(`Detected file change for ${path} in library ${library.id}`);
           if (matcher(path)) {
             // Note: if the changed file was not previously imported, it will be imported now.
             await this.scanAssets(library.id, [path], library.ownerId, false);
           }
-          this.emit('change', path);
+          this.emit(StorageEventType.CHANGE, path);
         },
         onUnlink: async (path) => {
           this.logger.debug(`Detected deleted file at ${path} in library ${library.id}`);
           const asset = await this.assetRepository.getByLibraryIdAndOriginalPath(library.id, path);
           if (asset && matcher(path)) {
             await this.assetRepository.save({ id: asset.id, isOffline: true });
           }
-          this.emit('unlink', path);
+          this.emit(StorageEventType.UNLINK, path);
         },
         onError: (error) => {
           // TODO: should we log, or throw an exception?
```

Comment:
Unrelated but I think this TODO is some artifact from an older PR that can be deleted

Reply:
Let's just keep it as logging an error and emitting that event